assets/src/blocks/Happypoint/HappypointEditor.js

```diff
@@ -0,0 +1,178 @@
+import { Fragment } from '@wordpress/element';
+import { HappypointFrontend } from './HappypointFrontend';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+import { debounce } from 'lodash';
+
+const {
+  InspectorControls,
+  BlockControls,
+  MediaUpload,
+  MediaUploadCheck
+} = wp.blockEditor;
+
+const { __ } = wp.i18n;
+
+import {
+  TextControl,
+  FocalPointPicker,
+  ToggleControl,
+  RangeControl,
+  PanelBody,
+  Button,
+  Toolbar,
+  IconButton
+} from '@wordpress/components';
+
+export const HappypointEditor = ({ attributes, setAttributes, isSelected }) => {
+  const { focus_image, opacity, mailing_list_iframe, id, iframe_url } = attributes;
+  const [iframeUrl, setIframeUrl] = useState(iframe_url || '');
+  const dimensions = { width: 400, height: 100 };
+
+  const { imageUrl } = useSelect(select => {
+    let imageUrl = '';
+    if (id && id > 0) {
+      const imageDetails = select('core').getMedia(id);
+      imageUrl = (imageDetails && imageDetails.source_url) || '';
+    }
+    return { imageUrl };
+  }, []);
```

Comment:
Maybe we should list `id`  as a dependency here, so it gets memoized and this hook is called again in case it changes, to make sure the `FocalPointPicker` using `imageURL` below is re-rendered (check my screenshots in the Conversation tab!).